app/src/lib/git/diff.ts

```diff
@@ -25,13 +24,56 @@ import { spawnAndComplete } from './spawn'
 import { DiffParser } from '../diff-parser'
 
 /**
+ * V8 has a limit on the size of string it can create, and unless we want to
+ * trigger an unhandled exception we need to do the encoding conversion by hand.
+ *
+ * This is a hard limit on how big a buffer can be and still be converted into
+ * a string.
+ */
+const MaxDiffBufferSize = 268435441
+
+/**
+ * Where `MaxDiffBufferSize` is a hard limit, this is a suggested limit. Diffs
+ * bigger than this _could_ be displayed but it might cause some slowness.
+ */
+const MaxReasonableDiffSize = 3000000
+
+/**
+ * The longest line length we should try to display. If a diff has a line longer
+ * than this, we probably shouldn't attempt it.
+ */
+const MaxLineLength = 500000
+
+/**
  * Utility function to check whether parsing this buffer is going to cause
  * issues at runtime.
  *
  * @param output A buffer of binary text from a spawned process
  */
-function isValidBuffer(output: Buffer) {
-  return output.length < maximumDiffStringSize
+function isValidBuffer(buffer: Buffer) {
+  return buffer.length < MaxDiffBufferSize
+}
+
+/** Is the buffer too large for us to reasonably represent? */
+function isBufferTooLarge(buffer: Buffer) {
+  if (!isValidBuffer(buffer)) {
+    return true
+  }
+
+  return buffer.length >= MaxReasonableDiffSize
```

Comment:
Could this be `isValidBuffer(buffer) && buffer.length < MaxReasonableDiffSize`?

Reply:
lol, I got those confused. I see you understood what I meant tho ✨

Reply:
Throw a De Morgan's Law on it 💥